dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/ConfigurablePeriod.java

```diff
@@ -63,26 +63,14 @@ public int hashCode()
     }
 
     @Override
-    public boolean equals( Object o )
+    public boolean equals( Object obj )
     {
-        if ( this == o )
-        {
-            return true;
-        }
-
-        if ( o == null )
-        {
-            return false;
-        }
-
-        if ( !(o instanceof Period) )
-        {
-            return false;
-        }
-
-        final Period other = (Period) o;
+        return this == obj || obj instanceof ConfigurablePeriod && objectEquals( (ConfigurablePeriod) obj );
+    }
 
-        return value.equals( other.getIsoDate() );
```

Comment:
@Lars Not quite sure if this is intentional or by accident. There is just one use of the `ConfigurablePeriod` which seams to just read. I assumed that at no point we would compare a `Period` object with a `ConfigurablePeriod`.

Reply:
I am not quite sure, but we use ConfigurablePeriod in analytics (BaseAnalyticalObject) so I am a bit hesitant to just take it out without understanding unintended effects.

Reply:
I reverted the equals semantics to the one that were present.